code/modules/mob/living/inhand_holder.dm

```diff
@@ -0,0 +1,123 @@
+//Generic system for picking up mobs.
+//Currently works for head and hands.
+/obj/item/clothing/head/mob_holder
+	name = "bugged mob"
+	desc = "Yell at coderbrush."
+	icon = null
+	icon_state = ""
+	var/mob/living/held_mob
+	var/can_head = FALSE
+	w_class = WEIGHT_CLASS_BULKY
+
+/obj/item/clothing/head/mob_holder/Initialize(mapload, mob/living/M, _worn_state, alt_worn, lh_icon, rh_icon, _can_head_override = FALSE)
+	. = ..()
+
+	if(M)
+		M.setDir(SOUTH)
+		held_mob = M
+		M.forceMove(src)
+		appearance = M.appearance
+		name = M.name
+		desc = M.desc
+
+	if(_can_head_override)
+		can_head = _can_head_override
+	if(alt_worn)
+		alternate_worn_icon = alt_worn
+	if(_worn_state)
+		item_state = _worn_state
+		icon_state = _worn_state
+	if(lh_icon)
+		lefthand_file = lh_icon
+	if(rh_icon)
+		righthand_file = rh_icon
+	if(!can_head)
+		slot_flags = NONE
+
+/obj/item/clothing/head/mob_holder/Destroy()
+	if(held_mob)
+		release()
+	return ..()
+
+/obj/item/clothing/head/mob_holder/dropped()
+	..()
+	if(isturf(loc))//don't release on soft-drops
+		release()
+
+/obj/item/clothing/head/mob_holder/proc/release(del_on_release = TRUE)
+	if(isliving(loc))
+		var/mob/living/L = loc
+		L.dropItemToGround(src)
+	if(held_mob)
+		var/mob/living/m = held_mob
+		m.forceMove(get_turf(m))
+		m.reset_perspective()
+		m.setDir(SOUTH)
+		held_mob = null
+	if(del_on_release)
+		qdel(src)
+
+/obj/item/clothing/head/mob_holder/relaymove(mob/user)
+	to_chat(user, "<span class='warning'>You must resist in order to escape!</span>")
+	return
+
+/obj/item/clothing/head/mob_holder/container_resist()
+	if(isliving(loc))
+		var/mob/living/L = loc
+		visible_message("<span class='warning'>[src] escapes [L]!</span>")
+	release()
+
+/mob/living/proc/mob_pickup(mob/living/L)
+	var/obj/item/clothing/head/mob_holder/holder = generate_mob_holder()
+	if(!holder) return
+	drop_all_held_items()
+	L.put_in_hands(holder)
+	return
+
+/mob/living/proc/generate_mob_holder()
+	..()
+	var/obj/item/clothing/head/mob_holder/holder = new(get_turf(src), src, (istext(can_be_held) ? can_be_held : ""), 'icons/mob/animals_held.dmi', 'icons/mob/animals_held_lh.dmi', 'icons/mob/animals_held_rh.dmi')
+	return holder
+
+/mob/living/simple_animal/drone/generate_mob_holder()
```

Comment:
I'd be able to sleep at night far better if all these `generate_mob_holder()` procs were in the relevant files all over the codebase.